Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -478,48 +593,87 @@ bool AzAssetBrowserRequestHandler::CanAcceptDragAndDropEvent(
+    // this default fallback handler will choose the most likely representitive one using a heuristic
+    // and only return that one.
+    // once AB supports multi-select, the data might contain a mixture of
+    // selected source files, selected products.  Selecting a source file should evaluate its products
+    // selecting a product should always use the product.
+
+    AZStd::vector<const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry*> *outProductsVector = nullptr;
+    if (outProducts.has_value())
+    {
+        outProductsVector = outProducts.value();
+    }
 
-    // Detects Product Assets that are dragged directly, or child Products of other entry types.
-    AzToolsFramework::AssetBrowser::AssetBrowserEntry::ForEachEntryInMimeData<ProductAssetBrowserEntry>(
-        event->mimeData(), [&](const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry* product)
+    bool canAcceptEvent = false;
+    AZStd::vector<const AssetBrowserEntry*> allEntries;
+    if (!AzToolsFramework::AssetBrowser::Utils::FromMimeData(mimeData, allEntries))
+    {
+        return false;
+    }
+    // all entries now contains the actual list of actually selected entries (not their children)
+    for (const AssetBrowserEntry* entry : allEntries)
     {
-        // Skip if this product is a child of a source file that is handled
-        if (outSources.has_value() && !outSources.value()->empty())
+        if (entry->GetEntryType() == AssetBrowserEntry::AssetEntryType::Source)
         {
-            auto parent = azrtti_cast<const SourceAssetBrowserEntry*>(product->GetParent());
-            if (parent != nullptr && AZStd::find(outSources.value()->begin(), outSources.value()->end(), parent) != outSources.value()->end())
+            // This default fallback handler doesn't care about source files.
+            // If you want to support source file drag operations, implement a listener with a higher priority
+            // in your gem, and connect to the drag and drop busses.  You can then intercept it and use code similar
+            // to the above to decode the mime data and do whatever you need to do, consuming the event so that
+            // we don't reach this point.
+
+            // Because of that, this handler only cares about products, so we convert any source entries to the most
+            // reasonable target product candidate.
+            AZStd::vector<const ProductAssetBrowserEntry*> candidateProducts;
+            entry->GetChildrenRecursively<ProductAssetBrowserEntry>(candidateProducts);
+            const ProductAssetBrowserEntry* mostReasonable = GetPrimaryProduct(candidateProducts);
+            if (mostReasonable)
             {
-                return;
+                canAcceptEvent = true;
+                if (outProductsVector)
+                {
+                    if (AZStd::find(outProductsVector->begin(), outProductsVector->end(), mostReasonable) == outProductsVector->end())
```

Comment:
FYI: [AZStd::ranges::find](https://en.cppreference.com/w/cpp/algorithm/ranges/find) functions are available for shorter typing.

```suggestion
                    if (AZStd::ranges::find(outProductsVector, mostReasonable) == outProductsVector->end())
```